sqlx-sqlite/src/index.rs

```diff
@@ -0,0 +1,567 @@
+                            SqlValue::Int(None).into(),
+                            SqlValue::Int(None).into(),
+                            min.into(),
+                            max.into(),
+                        ],
+                }})
+                .build_sqlx(SqliteQueryBuilder);
+            sqlx::query_with(&sql, values).execute(&mut *transaction).await?;
+        }
+
+        transaction.commit().await?;
+
+        Ok(())
+    }
+
+    /// Simple migration function that idempotently creates the table for the index
+    pub async fn initialize(&self) -> anyhow::Result<()> {
+        let query = sea_query::Table::create()
+            .table(FileStatistics::Table)
+            .if_not_exists()
+            .col(sea_query::ColumnDef::new(FileStatistics::FileId).big_integer().auto_increment().primary_key())
+            .col(sea_query::ColumnDef::new(FileStatistics::FileName).string().not_null().unique_key())
+            .col(sea_query::ColumnDef::new(FileStatistics::FileSizeBytes).big_integer().not_null())
+            .col(sea_query::ColumnDef::new(FileStatistics::RowGroupCount).big_integer().not_null())
+            .col(sea_query::ColumnDef::new(FileStatistics::RowCount).big_integer().not_null())
+            .build(SqliteQueryBuilder);
+
+        sqlx::query(&query).execute(&self.pool).await?;
+
+        let query = sea_query::Table::create()
+            .table(ColumnStatistics::Table)
+            .if_not_exists()
+            .col(sea_query::ColumnDef::new(ColumnStatistics::FileId).big_integer().not_null())
+            .col(sea_query::ColumnDef::new(ColumnStatistics::ColumnName).string().not_null())
+            .col(sea_query::ColumnDef::new(ColumnStatistics::RowGroup).big_integer().not_null())
+            .col(sea_query::ColumnDef::new(ColumnStatistics::NullCount).big_integer())
+            .col(sea_query::ColumnDef::new(ColumnStatistics::RowCount).big_integer())
+            .col(sea_query::ColumnDef::new(ColumnStatistics::IntMinValue).big_integer())
+            .col(sea_query::ColumnDef::new(ColumnStatistics::IntMaxValue).big_integer())
+            .col(sea_query::ColumnDef::new(ColumnStatistics::StringMinValue).string())
+            .col(sea_query::ColumnDef::new(ColumnStatistics::StringMaxValue).string())
+            .build(SqliteQueryBuilder);
+
+        sqlx::query(&query).execute(&self.pool).await?;
+
+        Ok(())
+    }
+}
+
+
+#[derive(Debug, Clone)]
+pub struct FileScanPlan {
+    pub file_size: u64,
+    pub access_plan: ParquetAccessPlan,
+}
+
+#[derive(Debug, Clone, Iden)]
+enum FileStatistics {
+    Table,
+    FileId,
+    FileName,
+    FileSizeBytes,
+    RowGroupCount,
+    RowCount,
+}
+
+#[derive(Debug, Clone, Iden)]
+enum ColumnStatistics {
+    Table,
+    FileId,
+    ColumnName,
+    RowGroup,
+    NullCount,
+    RowCount,
+    IntMinValue,
+    IntMaxValue,
+    StringMinValue,
+    StringMaxValue,
+    // Extend with other types as needed
```

Comment:
Although I'm not testing any string columns I thought it was worth sketching out how you would support arbitrary columns with arbitrary types. SQLite is also limited in the column types it supports but I think it would be easy to extend this to support timestamps, etc. with postgres or another database.